lib/rust/ensogl/component/text/src/font/msdf/src/lib.rs

```diff
@@ -0,0 +1,406 @@
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+// === Non-Standard Linter Configuration ===
+#![allow(clippy::option_map_unit_fn)]
+#![allow(clippy::precedence)]
+#![allow(dead_code)]
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use crate::prelude::*;
+use binding::*;
+
+use emscripten_data::ArrayMemoryView;
+use enso_prelude::FallibleResult;
+use js_sys::Uint8Array;
+use owned_ttf_parser::Tag;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use wasm_bindgen::prelude::Closure;
+use wasm_bindgen::JsValue;
+
+mod binding;
+pub mod emscripten_data;
+pub mod texture;
+
+pub use texture::*;
+
+/// Common types.
+pub mod prelude {
+    pub use enso_prelude::*;
+    pub use enso_types::*;
+}
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Fail, Eq, PartialEq)]
+pub enum SetVariationAxisError {
+    #[fail(display = "Invalid axis name: should fit in 4 bytes.")]
+    InvalidAxisName { name: String },
+    #[fail(
+        display = "Msdfgen `setVariationAxis` operation was not successfull for axis: {}.",
+        name
+    )]
+    LibraryError { name: String },
+}
+
+
+
+// ======================
+// === Initialization ===
+// ======================
+
+/// Add initialization callback. The callback passed as argument will be called once the msdfgen
+/// library is initialized.
+pub fn run_once_initialized<F>(callback: F)
+where F: 'static + FnOnce() {
+    if is_emscripten_runtime_initialized() {
+        callback()
+    } else {
+        let js_callback = Closure::once_into_js(callback);
+        on_emscripten_runtime_initialized(js_callback);
+    }
+}
+
+/// A future which resolves once the msdfgen library is initialized.
+pub fn initialized() -> impl Future<Output = ()> {
+    MsdfgenJsInitialized()
+}
+
+/// A future for running test after initialization.
+#[derive(Debug)]
+struct MsdfgenJsInitialized();
+
+impl Future for MsdfgenJsInitialized {
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        if is_emscripten_runtime_initialized() {
+            Poll::Ready(())
+        } else {
+            let waker = cx.waker().clone();
+            run_once_initialized(move || waker.wake());
+            Poll::Pending
+        }
+    }
+}
+
+
+
+// =================
+// === OwnedFace ===
+// =================
+
+/// A font face loaded to JS memory and handled by the msdfgen library.
+///
+/// The name uses the "owned" prefix in order to stay compatible with the `ttf-parser` library. In
+/// most cases, non-owned version is desirable, however, as this library uses JS interop (which is
+/// costly), keeping the owned version makes more sense.
```

Comment:
So what exactly "Owned" means? Why non-owned version is desirable? I miss a short explanation or at least a link.

Reply:
This sentence was misleading, you are right. Non-owned is like `&str`. It is not "desirable". Removing this sentece.